pkg/txn/storage/txn/batch.go

```diff
@@ -20,230 +20,164 @@ import (
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_int8:
-				col := vec.Col.([]int8)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_int16:
-				col := vec.Col.([]int16)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_int32:
-				col := vec.Col.([]int32)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_int64:
-				col := vec.Col.([]int64)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_uint8:
-				col := vec.Col.([]uint8)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_uint16:
-				col := vec.Col.([]uint16)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_uint32:
-				col := vec.Col.([]uint32)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_uint64:
-				col := vec.Col.([]uint64)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_float32:
-				col := vec.Col.([]float32)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_float64:
-				col := vec.Col.([]float64)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_tuple:
-				col := vec.Col.([][]any)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_char, types.T_varchar, types.T_json, types.T_blob:
-				if i < vec.Length() {
-					tuple = append(tuple, vec.GetBytes(int64(i)))
-				} else {
-					return
-				}
-
-			case types.T_date:
-				col := vec.Col.([]types.Date)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_datetime:
-				col := vec.Col.([]types.Datetime)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_timestamp:
-				col := vec.Col.([]types.Timestamp)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_decimal64:
-				col := vec.Col.([]types.Decimal64)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			case types.T_decimal128:
-				col := vec.Col.([]types.Decimal128)
-				if i < len(col) {
-					tuple = append(tuple, col[i])
-				} else {
-					return
-				}
-
-			}
+			value, isNull := vectorAt(vec, i)
+			tuple = append(tuple, value)
+			isNulls = append(isNulls, isNull)
 		}
-
 		i++
-
 		return
 	}
-
 	return iter
 }
 
-func vectorAt(vec *vector.Vector, i int) any {
+func vectorAt(vec *vector.Vector, i int) (value any, isNull bool) {
```

Comment:
Is that possible to remove use of any?   Any cause such a long switch because generics cannot kick in.

Reply:
If vectorAt is generic, I have to do the type switch at the call site. Unless the Vector is generic, the long switch is hard to avoid.